nixos/modules/services/databases/postgresql.nix

```diff
@@ -28,8 +28,86 @@ in
 
   ###### interface
 
-  options = {
-
+  options = let
+    databaseOpts = {
+      options = {
+        name = mkOption {
+          type = types.str;
+          description = ''
+            The name of the database to ensure.
+          '';
+          example = "gitlab";
+        };
+        owner = mkOption {
+          type = with types; nullOr str;
+          default = null;
+          description = ''
+            The name of the database owner.
+
+            The user will be created if it doesn't already
+            exist. The owner of an existing database will be
+            changed if it's changed here, and all objects in the
+            database owned by the previous owner will be
+            reassigned to the new one.
+
+            See
+            <link xlink:href="https://www.postgresql.org/docs/current/manage-ag-createdb.html">Creating a Database</link>,
+            <link xlink:href="https://www.postgresql.org/docs/current/sql-reassign-owned.html">REASSIGN OWNED</link>
+            and
+            <link xlink:href="https://www.postgresql.org/docs/current/sql-alterdatabase.html">ALTER DATABASE</link>
+            for more details.
+          '';
+        };
+        extensions = mkOption {
```

Comment:
This option seems to be on the wrong level.
Extensions in Postgres do not exist on a database level.
In fact, you cannot even have the same extension loaded in different versions on different schemas, even though the documentation might suggest that.
Because of that, I think this option belongs on the service layer, not on a database specific layer in the options.

Reply:
Hm, that doesn't sound right. I'd have to test this myself to confirm, but if that's true, the documentation is plain wrong. Refer to https://www.postgresql.org/docs/current/sql-createextension.html - the first sentence is `CREATE EXTENSION loads a new extension into the current database.`

Reply:
It's not impossible I am remembering this from an older version, but I have this nagging feeling that something was iffy here. Best to test, I agree.